### PR TITLE
Fixed failing test cases for TCP transmission unit tests

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR fixes the failing unit tests for TCP transmission unit tests and fixes the build issue with UDP IP unit tests after merge of new changes as part of the IPv6 integration.

Test Steps
-----------

```
cmake -S test/unit-test -B test/unit-test/build/ 
make -C test/unit-test/build/ all 
cd test/unit-test/build/
ctest -E system --output-on-failure --verbose

```
Related Issue
-----------
Failing TCP transmission test cases.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/CMakeLists.txt [hunk 1]
```diff
@@ -127,8 +127,8 @@ foreach( file ${TCP_INCLUDES} )
     endif()
 
     # Use this tool to process all conditional declarations.
-    if(${MODIFIED_FILE} STREQUAL "FreeRTOS_Routing" )
-        execute_process( COMMAND unifdefall -U${Guard} -USEND_REPEATED_COUNT -UpdTRUE_SIGNED -UFreeRTOS_htonl -D__COVERITY__ -DTEST -DipconfigUSE_IPV6 -I ${MODULE_ROOT_DIR}/tools/CMock/vendor/unity/src
+    if(${MODIFIED_FILE} STREQUAL "FreeRTOS_Routing" OR ${MODIFIED_FILE} STREQUAL "FreeRTOS_IP_Private" )
```
Comment (moninom1) [inline test/unit-test/CMakeLists.txt:130]: Why is this needed?

File: test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c [hunk 2]
```diff
@@ -88,63 +88,76 @@ uint8_t ucEthernetBuffer[ ipconfigNETWORK_MTU ] =
 };
 
 
-/* test for prvTCPMakeSurePrepared function */
-void test_prvTCPMakeSurePrepared_Not_Ready( void )
-{
-    BaseType_t xResult = 0;
+// /* test for prvTCPMakeSurePrepared function */
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c:91]: Why is this commented?

File: test/unit-test/CMakeLists.txt (hunk 1 above)
Comment (tony-josi-aws) [reply test/unit-test/CMakeLists.txt:130]: `FreeRTOS_Routing.h` and `FreeRTOS_IP_Private.h` needs to be parsed with `ipconfigUSE_IPV6 `and `ipconfigUSE_RA` enabled in order to make certain structures defined inside those headers compatible with the current test case requirements.

File: test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c (hunk 2 above)
Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c]: Didnt uncomment by mistake, updated.

File: test/unit-test/ConfigFiles/FreeRTOSIPConfig.h [hunk 3]
```diff
@@ -36,8 +36,6 @@
 
 #define ipconfigUSE_IPV4            ( 1 )
 
-#define ipconfigUSE_IPV6            ( 1 )
-
```
Comment (shubnil) [inline test/unit-test/ConfigFiles/FreeRTOSIPConfig.h:40]: Do we not need ipconfigUSE_IPV6 flag here

File: test/unit-test/FreeRTOS_ARP_DataLenLessThanMinPacket/FreeRTOS_ARP_DataLenLessThanMinPacket_stubs.c [hunk 4]
```diff
@@ -148,84 +148,8 @@ void vApplicationGetIdleTaskMemory( StaticTask_t ** ppxIdleTaskTCBBuffer,
...
-{
-}
-/*
- * Get the next Network Interface.
- */
-NetworkInterface_t * FreeRTOS_NextNetworkInterface( const NetworkInterface_t * pxInterface )
-{
-}
\ No newline at end of file
+void FreeRTOS_OutputAdvertiseIPv6( NetworkEndPoint_t * pxEndPoint ) {}
```
Comment (shubnil) [inline test/unit-test/FreeRTOS_ARP_DataLenLessThanMinPacket/FreeRTOS_ARP_DataLenLessThanMinPacket_stubs.c:155]: Please explain the reason of having this blank function

File: test/unit-test/ConfigFiles/FreeRTOSIPConfig.h (hunk 3 above)
Comment (tony-josi-aws) [reply test/unit-test/ConfigFiles/FreeRTOSIPConfig.h:40]: we are not globally enabling ipv6 flag, instead only for tests that require ipv6 flags to make it build

File: test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c [hunk 5]
```diff
@@ -407,21 +451,30 @@ void test_prvTCPReturnPacket_Null_Socket( void )
     usGenerateChecksum_ExpectAnyArgsAndReturn( 0x1111 );
     usGenerateProtocolChecksum_ExpectAnyArgsAndReturn( 0x2222 );
     eARPGetCacheEntry_ExpectAnyArgsAndReturn( eARPCacheHit );
-    xNetworkInterfaceOutput_ExpectAnyArgsAndReturn( pdTRUE );
+    //xNetworkInterfaceOutput_ExpectAnyArgsAndReturn( pdTRUE );
```
Comment (shubnil) [inline test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c:454]: Please remove this if it's a debug leftovers.

File: test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c [hunk 6]
```diff
@@ -436,11 +489,12 @@ void test_prvTCPReturnPacket_Null_Socket_Relase_True( void )
...
```
Comment (shubnil) [inline test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c:492]: Please remove this if it's a debug leftovers.

File: test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c [hunk 7]
```diff
@@ -602,13 +682,14 @@ void test_prvTCPReturnPacket_No_KL_LocalIP_ARP_Not_Hit( void )
 
     uxIPHeaderSizePacket_IgnoreAndReturn(ipSIZE_OF_IPv4_HEADER);
     uxStreamBufferFrontSpace_ExpectAnyArgsAndReturn( 1000 );
+    //FreeRTOS_FindEndPointOnNetMask_ExpectAnyArgsAndReturn(&xEndPoint);
```
Comment (shubnil) [inline test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c:685]: Please remove this if it's a debug leftovers.

**Review by shubnil — COMMENTED** [review]
Added some comments and clarifications.

File: test/unit-test/FreeRTOS_ARP_DataLenLessThanMinPacket/FreeRTOS_ARP_DataLenLessThanMinPacket_stubs.c (hunk 4 above)
Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_ARP_DataLenLessThanMinPacket/FreeRTOS_ARP_DataLenLessThanMinPacket_stubs.c:155]: Thats used as  a stub to fix the build failure with undefined function

File: test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c (hunk 5 above)
Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c:454]: sure, will update that

Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_TCP_Transmission/FreeRTOS_TCP_Transmission_utest.c]: fixed all

**Review by shubnil — APPROVED** [review]
